target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4018-whw01-v1.dts

```diff
@@ -0,0 +1,237 @@
+	};
+
+	chosen {
+		bootargs-append = " root=/dev/ubiblock0_0";
+	};
+
+	reserved-memory {
+		rsvd1@87000000 {
+			reg = <0x87000000 0x500000>;
+			no-map;
+		};
+
+		wifi_dump@87500000 {
+			reg = <0x87500000 0x600000>;
+			no-map;
+		};
+
+		rsvd2@87B00000 {
+			reg = <0x87b00000 0x300000>;
+			no-map;
+		};
+	};
+
+	soc {
+		gpio_keys {
+			pinctrl-0 = <0x52>;
+			pinctrl-names = "default";
+			compatible = "gpio-keys";
+
+			button@1 {
+				label = "reset";
+				linux,code = <0x198>;
+				gpios = <&tlmm 0x3f 0x01>;
+				linux,input-type = <0x01>;
+			};
+		};
+	};
+};
+
+&tlmm {
+	i2c_0_pins: i2c_0_pinmux {
+		mux {
+			function = "blsp_i2c0";
+			pins = "gpio58", "gpio59";
+			bias-disable;
+		};
+	};
+
+	reset_pinmux {
+		linux,phandle = <0x52>;
+		phandle = <0x52>;
+
+		mux {
+			pins = "gpio63";
+			bias-pull-up;
+		};
+	};
+};
+
+&spi_0_pins {
+	pinmux_cs {
+		pins = "gpio54", "gpio4";
+	};
+	pinconf_cs {
+		pins = "gpio54", "gpio4";
+	};
+};
+
+&blsp1_spi1 {
+	status = "okay";
+	pinctrl-0 = <&spi_0_pins>;
+	pinctrl-names = "default";
+	cs-gpios = <&tlmm 54 GPIO_ACTIVE_HIGH>, <&tlmm 4 GPIO_ACTIVE_HIGH>;
+
+	nor@0 {
+		spi-max-frequency = <0x16e3600>;
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <0x01>;
+			#size-cells = <0x01>;
+
+			partition@0 {
+				label = "0:SBL1";
+				reg = <0x00 0x40000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "0:MIBIB";
+				reg = <0x40000 0x20000>;
+				read-only;
+			};
+
+			partition@60000 {
+				label = "0:QSEE";
+				reg = <0x60000 0x60000>;
+				read-only;
+			};
+
+			partition@c0000 {
+				label = "0:CDT";
+				reg = <0xc0000 0x10000>;
+				read-only;
+			};
+
+			partition@e0000 {
+				label = "APPSBL";
+				reg = <0xd0000 0xa0000>;
+				read-only;
+			};
+
+			partition@160000 {
+				label = "0:ART";
+				reg = <0x170000 0x10000>;
+				read-only;
+			};
+
+			partition@170000 {
+				label = "u_env";
+				reg = <0x180000 0x40000>;
+			};
+
+			partition@190000 {
+				label = "s_env";
+				reg = <0x1c0000 0x20000>;
+			};
+
+			partition@1b0000 {
+				label = "devinfo";
+				reg = <0x1e0000 0x20000>;
+			};
+		};
+	};
+
+	nand@1 {
+		reg = <1>;
+		status = "ok";
+		compatible = "spi-nand";
+		spi-max-frequency = <0x16e3600>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <0x01>;
+			#size-cells = <0x01>;
+
+			partition@0 {
+				label = "kernel";
+				reg = <0x00 0x5000000>;
+				read-only;
+			};
+
+			partition@600000 {
+				label = "rootfs";
+				reg = <0x600000 0x4a00000>;
+			};
+
+			partition@5000000 {
+				label = "alt_kernel";
+				reg = <0x5000000 0x5000000>;
+				read-only;
+			};
+
+			partition@5600000 {
+				label = "alt_rootfs";
+				reg = <0x5600000 0x4a00000>;
+			};
+
+			partition@a000000 {
+				label = "sysdiag";
+				reg = <0xa000000 0x200000>;
+			};
+
+			partition@a200000 {
+				label = "syscfg";
+				reg = <0xa200000 0x5e00000>;
+			};
```

Comment:
`sysdiag` appears to be empty on all three of my test devices (filled with 0xFF), the usage unknown. I'll re-add for `sysdiag`.

`syscfg` is a UBIFS and contains a number of backup files and SQLite databases used by the factory firmware - such as databases of all MAC addresses ever seen by the device, etc. This is recreated if I flash back the factory firmware, so allowing RW allows the end-user to replace / re-use as desired - or at least remove the SQLite databases which contain device information about clients which persist between upgrades.

Reply:
Well, if it's factory stuff I would say it should not be altered by the OpenWrt user. However, you saying "it's recreated" means whatever you do here with OpenWrt, it will be wiped when installing OEM firmware anyway?

Reply:
That's correct. The factory firmware recreates the contents of this UBIFS. If it **does** exist, it continues to use the database files on the volume between firmware versions. These SQLite databases are used for what could only be described as "long lived telemetry" about devices seen by the units.